### DISCUSSION TITLE
Overlay networking VPN

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [goastian/midori-desktop#41] [Ideas]
* midori does not need to pay for VPN servers any more.

overlay networking since each node is both a server and a client
overlay networking to create a direct encrypted tunnel between two nodes even if they’re both behind firewalls. Each node doesn’t even need to know the other’s IP address before starting the connection.
* idea 

CDN, Website, ActivityPub,  Fediverse, Router, Android, iOS, Linux, BSD, "Domain generation algorithm" "Fast flux"  **All use overlay networking**
Allow users in censored countries to use your network connection to access the network.
Help people in censored countries bypass censorship by running the

[Hyprspace](https://github.com/hyprspace/hyprspace#hyprspace)
[n2n](https://github.com/ntop/n2n#n2n)
[nebula](https://github.com/slackhq/nebula)
[qaul](https://github.com/qaul/qaul.net)

P2P
[Hypercore Protocol](https://github.com/hypercore-protocol) and [pinecone](https://github.com/matrix-org/pinecone) and [libp2p](https://libp2p.io/) and [devp2p](https://github.com/ethereum/devp2p)
bittorrent
WebRTC
[gemini](https://gemini.circumlunar.space/)
[Earthstar](https://earthstar-project.org/)
[Pigeon](https://tildegit.org/PigeonProtocolConsortium/Protocol-Spec)
[geneva Protocol](https://github.com/Kkevsterrr/geneva)

**Comment by Ponchale** [comment]
interesting

**Comment by Ponchale** [comment]
> * midori does not need to pay for VPN servers any more.
> 
> 
> overlay networking since each node is both a server and a client overlay networking to create a direct encrypted tunnel between two nodes even if they’re both behind firewalls. Each node doesn’t even need to know the other’s IP address before starting the connection. [Hyprspace](https://github.com/hyprspace/hyprspace#hyprspace) [n2n](https://github.com/ntop/n2n#n2n) [nebula](https://github.com/slackhq/nebula) [qaul](https://github.com/qaul/qaul.net)
> 
> P2P [Hypercore Protocol](https://github.com/hypercore-protocol) and [pinecone](https://github.com/matrix-org/pinecone) and [libp2p](https://libp2p.io/) and [devp2p](https://github.com/ethereum/devp2p) WebRTC [gemini](https://gemini.circumlunar.space/) [Earthstar](https://earthstar-project.org/) [Pigeon](https://tildegit.org/PigeonProtocolConsortium/Protocol-Spec) [geneva Protocol](https://github.com/Kkevsterrr/geneva)

Friend, do we know the standards that these services have implemented?

**Comment by Ponchale** [comment]
We started to make our own adaptation of wireguard
